### PR TITLE
remove usermap by name, introduced more optionals

### PR DESCRIPTION
**Pull request by ingmargoudt**
I removed the map usersByName from the UserManager. This class had two maps of users, sorted by id, and sorted by name. I think it is sufficient to stick with just one map. 
Next to that, I introduced more optionals where possible.

### CODE REVIEW

File: Mage.Server/src/main/java/mage/server/ChatManager.java [hunk 1]
```diff
@@ -157,10 +158,9 @@ public void broadcast(UUID chatId, String userName, String message, MessageColor
                     }
 
                     userMessages.put(userName, message);
-                }
 
-                if (messageType == MessageType.TALK) {
-                    if (user != null) {
```
Comment (drmDev) [inline Mage.Server/src/main/java/mage/server/ChatManager.java]: I see you removed a check for user being null here but no corresponding user.isPresent(). Is that intentional?

File: Mage.Server/src/main/java/mage/server/MageServerImpl.java [hunk 2]
```diff
@@ -1071,28 +1071,26 @@ public void disconnectUser(final String sessionId, final String userSessionId) t
     @Override
     public void muteUser(final String sessionId, final String userName, final long durationMinutes) throws MageException {
         execute("muteUser", sessionId, () -> {
-            User user = UserManager.instance.getUserByName(userName);
-            if (user != null) {
```
Comment (drmDev) [inline Mage.Server/src/main/java/mage/server/MageServerImpl.java]: same idea here as above - perhaps I don't understand the implications of changing to Java optionals in terms of null ref exceptions when not checking on isPresent

File: Mage.Server/src/main/java/mage/server/MageServerImpl.java [hunk 3]
```diff
@@ -1144,13 +1141,10 @@ public void endUserSession(final String sessionId, final String userSessionId) t
     @Override
     public void removeTable(final String sessionId, final UUID tableId) throws MageException {
         execute("removeTable", sessionId, () -> {
-            Optional<Session> session = SessionManager.instance.getSession(sessionId);
-            if (!session.isPresent()) {
-                logger.error("Session not found : " + sessionId);
```
Comment (drmDev) [inline Mage.Server/src/main/java/mage/server/MageServerImpl.java]: this section removed or just handled elsewhere? not sure how important this logging was but feel like it should be retained if we're not sure.

Comment (ingmargoudt) [inline Mage.Server/src/main/java/mage/server/MageServerImpl.java]: its handled in getSession()

File: Mage.Server/src/main/java/mage/server/ChatManager.java (hunk 1 above)
Comment (ingmargoudt) [inline Mage.Server/src/main/java/mage/server/ChatManager.java]: That check is already done, this line is inside the scope of the isPresent() on line#120

**Pull request merged**